scipy/cluster/hierarchy.py

```diff
@@ -1395,6 +1400,208 @@ def cut_tree(Z, n_clusters=None, height=None):
+        and so on. Also note that in each cluster splitting, the label 0
+        denotes the bigger cluster, while the label 1 denotes the smallest.
+
+    See Also
+    --------
+    cut_tree
+
+    Notes
+    -----
+    There are several implemented methods following the same idea, i.e.
+    performing a tree cut in which the resulting clusters are at different tree
+    levels, but using more elaborated algorithms (in which the threshold of
+    max_cluster_size is dinamycally computed). The CRAN R package
+    [dynamicTreeCut](https://github.com/cran/dynamicTreeCut) implements novel
+    dynamic branch cutting methods for detecting clusters in a dendrogram
+    depending on their shape. Further, [MLCut]](https://github.com/than8/MLCut)
+    provides interactive methods to cut tree branches at multiple levels.
+    Note that in the present method, the max_cluster_size threshold is a fixed
+    value given as input.
+
+    Further, note that this algorithm uses :math:`O(n^2)` memory, i.e. the same
+    as cut_tree because a full cut tree of the linkage matrix is performed
+    as the beginning. Further, this data structure is used in order to perform
+    the successive computations.
+
+
+    Examples
+    --------
+    >>> from scipy.cluster import hierarchy
+    >>> from scipy import stats
+
+    Initialize the random seed.
+
+    >>> np.random.seed(14)
+
+    Create a input matrix containing 100 data samples with 4 dimensions.
+    Note: using `gamma` in order to generate an unbalanced distribution.
+    If a regular ``cut_tree()`` would be performed, one big and many small
+    clusters would be obtained.
+
+    >>> X = stats.gamma.rvs(0.1, size=400).reshape((100, 4))
+
+    Compute the linkage matrix using the scipy ward() or linkage() method:
+
+    >>> Z = hierarchy.ward(X)
+
+    Perform a balanced cut tree of the linkage matrix:
+
+    >>> cluster_id, cluster_level = hierarchy.cut_tree_balanced(
+    ...                                              Z, max_cluster_size=10)
+    >>> cluster_id[:10]
+    array([19,  4, 10, 12, 20, 12, 14,  9, 15,  2])
+    >>> cluster_level[:10]
+    array([array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 0, 1]),
+           array([0, 0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]),
+           array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]),
+           array([0, 0, 1, 0])], dtype=object)
+
+    Note that clusters with more similar values for cluster_level denote
+    clusters with less distance in between, thus representing vectors which are
+    closer in the multidimensional space. This information contained within
+    cluster_level is not usually present in cluster_id.
```

Comment:
Also in this paragraph, double backticks around variable names with underscores

Reply:
Fixed (will be pushed when the current CI job is over).

Reply:
> will be pushed when the current CI job is over

not quite what I meant - if you are done with fixes for now you can push them. that will actually stop the previous CI run and start a new one, so uses less resources than letting it finish first and then pushing.

Reply:
Fixes are now pushed. Many thanks and best regards.